core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -587,10 +595,50 @@ public void callOptionsExecutor() {
   @Test
   public void nameResolutionFailed() {
     Status error = Status.UNAVAILABLE.withCause(new Throwable("fake name resolution error"));
+    FakeNameResolverFactory nameResolverFactory = new FakeNameResolverFactory(error);
+    // Name resolution is started as soon as channel is created.
+    createChannel(nameResolverFactory, NO_INTERCEPTOR);
+    FakeNameResolverFactory.FakeNameResolver resolver = nameResolverFactory.resolvers.get(0);
+    verify(mockLoadBalancer).handleNameResolutionError(same(error));
 
+    timer.forwardNanos(RECONNECT_BACKOFF_INTERVAL_NANOS);
+
+    assertEquals(1, resolver.refreshCalled);
+    verify(mockLoadBalancer, times(2)).handleNameResolutionError(same(error));
+
```

Comment:
To verify that you will not schedule duplicate timers, maybe also call refresh() here, and verify `handleNameResolutionError()` is called twice, and there is still one timer scheduled.

Reply:
Done